examples/scaling-criteo/02-ETL-with-NVTabular.ipynb

```diff
@@ -274,7 +274,11 @@
     "\n",
     "# Deploy a Single-Machine Multi-GPU Cluster\n",
     "protocol = \"tcp\"  # \"tcp\" or \"ucx\"\n",
-    "NUM_GPUS = [0, 1, 2, 3, 4, 5, 6, 7]\n",
+    "NUM_GPUS = os.environ.get(\"CUDA_VISIBLE_DEVICES\", None)\n",
+    "if not NUM_GPUS:\n",
+    "    NUM_GPUS = [0, 1, 2, 3, 4, 5, 6, 7]\n",
+    "else:\n",
+    "    NUM_GPUS = [int(num) for num in NUM_GPUS.split(',')]\n",
```

Comment:
I'm not sure this is quite right. Like if we set CUDA_VISIBLE_DEVICES=2,3 - then we only have 2 GPU's available , but the device ID's will be 0,1 in the current process not 2,3.

How about replacing this with something like

```
NUM_GPUS = list(range(cupy.cuda.runtime.getDeviceCount()))
```

I think this will respect the CUDA_VISIBLE_DEVICES environment, and also scales down to systems (like mine) that only have 2 gpu's in them.